mythril_multiboot2/src/services.rs

```diff
@@ -37,4 +37,35 @@ impl VmServices for Multiboot2Services {
     fn acpi_addr(&self) -> Result<HostPhysAddr> {
         Ok(HostPhysAddr::new(0))
     }
+
+    fn rsdp(&self) -> Result<crate::acpi::rsdp::RSDP> {
+        // Panic if can find a multiboot rsdpv2 tag
+        match self.info.rsdp_v2_tag() {
+            Some(_) => panic!("Uh oh: Found multiboot rsdp_v2_tag!"),
+            None => {},
+        } 
+
+        // Get a v1 tag if possible
+        match self.info.rsdp_v1_tag() {
+            Some(tag) => {
+                let oem_id = tag.oem_id().map(|found| found.as_bytes());
+                match oem_id {
+                    Some(id) => {
+                        let mut arr: [u8; 6] = [0; 6];
+                        arr.copy_from_slice(&id[0..6]);
+                        return Ok(crate::acpi::rsdp::RSDP::V1 {
+                            oemid: arr,
+                            rsdt_addr: tag.rsdt_address() as u32,
+                        });
+                    },
+                    None => {}
+                }
+            },
+            None => {}
+        }
```

Comment:
I think this could be written something like:
```
// Get a v1 tag if possible
        self.info.rsdp_v1_tag()
            .map(|tag| {
                // Use the empty string if no OEM id is provided
                let id = tag.oem_id().unwrap_or("      ").as_bytes();
                let mut arr: [u8; 6] = [0; 6];
                arr.copy_from_slice(&id[0..6]);
                crate::acpi::rsdp::RSDP::V1 {
                    oemid: arr,
                    rsdt_addr: tag.rsdt_address() as u32,
                }
            })
            .ok_or(Error::NotFound)
```
The main functional change here is that we don't bail out if there is no OEM ID. I don't think that should be an error condition if we actually _do_ have a RSDPv1 tag. But maybe the lack of the oem id indicates the RSDP is invalid in some way? Do you have insight here @dlrobertson ?

Reply:
https://github.com/rust-osdev/multiboot2-elf64/blob/cb0284e21649ef034d59948be0e439e6ea13f139/src/rsdp.rs#L29

An `oem_id` of `None` is an invalid utf8 sequence. Is there a OEM ID spec? I don't have much of an opinion here. As long as the checksum is valid the RSDP is valid, so In theory I could see a scenario where there is an invalid `oem_id`, but a valid RSDP... maybe we could have a default value that we populate the oem_id if it is none?

Reply:
Yeah, the snippet I posted uses six spaces as the default, which seems reasonable. I'm willing to bet there is some broken board out there that uses invalid utf-8 in their oem_id.

Reply:
done